tumblr/kanvas-ios
# Add Multiple Export

## Testing

- Install Example app.
- Enable the `Metal Filters`, `Camera Metal` , `Multi Export` features and disable the `Export Stop Motion as Video` feature.
- Add multiple photos and videos by tapping the new frame button in the bottom right.
- Ensure that the photos and videos are exported and saved back to Photos.

## Code

- `MediaArchiver.swift` contains the logic to save media (images or video) to disk.
- `MultiEditorExportHandler.swift` handles the multiple callbacks and collecting all of returned media/errors.
- API change to `didCreateMedia` to pass back multiple media items and errors instead of just one:

```diff
- func didCreateMedia(_ cameraController: CameraController, media: KanvasCameraMedia?, exportAction: KanvasExportAction, error: Error?)
+ func didCreateMedia(_ cameraController: CameraController, media: [(KanvasCameraMedia?, Error?)], exportAction: KanvasExportAction)
```

- Adds a `CameraController.saveDirectory` which is used to save the original, unmodified image to be reused later. This is necessary for stories so that we can pull the original image used from the story without modifications for later editing. It is optional and is not specified by the example app.

## Notes

- Media may be stretched due to the Metal renderer. This is expected for now.
